File: js/struc/histo.js
const http = require('../http/index.js')

const HistoricalWinstreaks = class {


  constructor(mode) {
    this.game = mode
  }

  lb(start, end) {
    var url
    if(start)
     url = `https://api.roccodev.pw/${this.game}/winstreaks/historical/leaderboard?from=${start}&to=${end}`
    else
     url = `https://api.roccodev.pw/${this.game}/winstreaks/historical/leaderboard`

    return http(url)
  }

  profile(uuid) {
    const url = `https://api.roccodev.pw/${this.game}/winstreaks/historical/profile/${uuid}`
    return http(url)
  }


}

module.exports = HistoricalWinstreaks

File: ruby/lib/histo.rb
require 'net/http'
require 'json'

class HistoricalWinstreaks

    def initialize(game)
      @game = game
    end

    def leaderboard(*args)
      if args.length == 2
        url = "https://api.rocco.dev/#{@game}/winstreaks/historical/leaderboard?from=#{args[0]}&to=#{args[1]}"
      else
        url = "https://api.rocco.dev/#{@game}/winstreaks/historical/leaderboard"
      end
      uri = URI(url)
      res = Net::HTTP.get(uri)
      return JSON.parse(res)
    end

    def profile(uuid)
      url = "https://api.rocco.dev/#{@game}/winstreaks/historical/profile/#{uuid}"
      uri = URI(url)
      res = Net::HTTP.get(uri)
      return JSON.parse(res)
    end

end
